jina/serve/streamer.py

```diff
@@ -221,3 +221,36 @@ def get_streamer():
     @staticmethod
     def _set_env_streamer_args(**kwargs):
         os.environ['JINA_STREAMER_ARGS'] = json.dumps(kwargs)
+
+
+class _ExecutorStreamer:
+    def __init__(self, connection_pool: GrpcConnectionPool, executor_name: str) -> None:
+        self._connection_pool: GrpcConnectionPool = connection_pool
+        self.executor_name = executor_name
+
+    async def post(
+        self,
+        inputs: DocumentArray,
+        request_size: int = 100,
+        on: Optional[str] = None,
+        parameters: Optional[Dict] = None,
+        **kwargs,
+    ):
+        reqs = []
+        for docs_batch in inputs.batch(batch_size=request_size, shuffle=False):
+            req = DataRequest()
+            req.header.exec_endpoint = on
+            req.header.target_executor = self.executor_name
+            req.parameters = parameters
+            req.data.docs = docs_batch
+            reqs.append(req)
+
+        resp, _ = await self._connection_pool.send_requests_once(
```

Comment:
do tests where u send more than one request, I believe this interface is not intended to be used like this.

Every `req` should lead to a `send_request` call and u should collect docs afterwards.

Reply:
ah ok done.